vpd-tool/src/vpd_tool_main.cpp

```diff
@@ -67,19 +88,48 @@ int main(int argc, char** argv)
         return l_rc;
     }
 
-    if (*l_hardwareFlag && !std::filesystem::exists(l_vpdPath))
```

Comment:
This is required for both read and write. So why are you moving it inside writteFlag implementation?

Reply:
currently hardware option is used for read and write operation.
so keeping this check where it can be provided, avoid any global validation checks unless not sure it may be used or not.